shank-render/src/pda/render_seeds.rs

```diff
@@ -0,0 +1,330 @@
+use quote::quote;
+use std::str::FromStr;
+
+use proc_macro2::{Ident, Span, TokenStream};
+use shank_macro_impl::{
+    parsed_struct::{
+        ProcessedSeed, Seed, ACCOUNT_INFO_TY, FULL_ACCOUNT_INFO_TY,
+        FULL_PUBKEY_TY, PUBKEY_TY,
+    },
+    syn::{Error as ParseError, Result as ParseResult},
+    types::{Composite, ParsedReference, Primitive, RustType, TypeKind, Value},
+};
+
+use super::render_args_comments;
+
+pub fn try_render_seeds_fn(
+    processed_seeds: &[ProcessedSeed],
+    seeds_fn_name: &Ident,
+    seeds_fn_with_bump_name: &Ident,
+    include_comments: bool,
+) -> ParseResult<Option<TokenStream>> {
+    let lifetime = "a";
+    let RenderedSeedsParts {
+        seed_array_items,
+        seed_fn_args,
+    } = try_render_seeds_parts(processed_seeds, lifetime)?;
+    if seed_array_items.is_empty() {
+        return Ok(None);
+    }
+
+    let len = seed_array_items.len();
+    let lifetime_toks = format!("<'{}>", lifetime).parse::<TokenStream>()?;
+    let len_with_bump = len + 1;
+    let bump = if seed_fn_args.is_empty() {
+        quote! { bump: &'a [u8; 1] }
+    } else {
+        quote! { , bump: &'a [u8; 1] }
+    };
+
+    let (seeds_comments, seeds_with_bump_comments) = if include_comments {
+        let args_comments = render_args_comments(processed_seeds, false);
+        (
+            format!(
+                r#"
+                /// Derives the seeds for this account.
+                ///
+                {}"#,
+                args_comments.join("\n")
+            )
+            .parse()
+            .unwrap(),
```

Comment:
```suggestion
            .to_token_stream()
```
might be better than `parse().unwrap()` ?